src/library.js

```diff
@@ -1766,19 +1766,27 @@ LibraryManager.library = {
   // setjmp.h
   // ==========================================================================
 
-#if SUPPORT_LONGJMP
   longjmp__sig: 'vii',
+#if SUPPORT_LONGJMP
   longjmp: function(env, value) {
     _setThrew(env, value || 1);
     throw 'longjmp';
   },
+#else
+  longjmp__deps: [function() {
+    error('longjmp support was disabled (SUPPORT_LONGJMP=0), but it is required by the code (either set SUPPORT_LONGJMP=0, or remove uses of it in the project)');
+  }],
+  // will never be emitted, as the dep errors at compile time
```

Comment:
Interesting approach, although a bit magical

Reply:
(the other option would be to add the check somewhere in `jsifier.js` - this way it's right alongside the relevant `longjmp` logic, which seems better to me)

Reply:
No I think this is fine.